### PR TITLE
Add acceptance tests for keyless workflow

### PR DESCRIPTION
**Pull request by lcarva**
https://issues.redhat.com/browse/HACBS-2071

An image signature/attesttaion created by cosign contains a SignedEntityTimestamp (SET). This includes all the information needed for verfication if the TUF root has already been initialized.

To initialize a TUF root, two parameters are needed. The TUF mirror and the trusted initial root.

The new acceptance tests leverage this property to only require the following:

1. An OCI image that has already been signed and attested by cosign.
2. WireMock recordings to capture the TUF mirror API calls required to initialize the TUF root.
3. The root.json representing the trusted initial root.

A script to generate all the required data is also included. This will be useful if we need to re-generate any of the data as needed. But it is not required as part of the regular development workflow.

### CODE REVIEW

**Review by zregvart — APPROVED** [review]
Cool, comments mostly nitpickery. Would be interesting to have `hack/generate-test-signed-images.sh` use the `hack/setup-dev-cluster.sh` cluster

File: acceptance/image/testimage/blobs/sha256/1f409d0e269308ac0b51b25b373fb062b02d2fc4e4a1e43a2d9462c29532fbe4 [hunk 1]
```diff
@@ -0,0 +1 @@
+{"created":"2023-04-27T13:06:06.703718333Z","architecture":"amd64","os":"linux","config":{"Labels":{"io.buildah.version":"1.30.0"}},"rootfs":{"type":"layers","diff_ids":["sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef"]},"history":[{"created":"2023-04-27T13:06:06.70823379Z","created_by":"/bin/sh"}]}
```
[the remark this answers is not in the document]
Comment (lcarva) [reply acceptance/image/testimage/blobs/sha256/1f409d0e269308ac0b51b25b373fb062b02d2fc4e4a1e43a2d9462c29532fbe4:1]: `createAndPushKeylessImage` loads these files into the registry. The confusion likely comes from the docs of `createAndPushKeylessImage` being incorrect. I'll update it.

File: hack/generate-test-signed-images.sh [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+# Check if the kind cluster already exists
+set +e
+kind get clusters 2> /dev/null | grep kind > /dev/null
+cluster_exists="$?"
+set -e
+if [[ "${cluster_exists}" -eq "0" ]]; then
+    echo '🛑 The kind cluster already exists. Delete it first:'
+    echo -e '\tkind delete cluster --name kind'
+    errors=true
+fi
```
[the remark this answers is not in the document]
Comment (lcarva) [reply hack/generate-test-signed-images.sh:57]: Yeah, that's why this is telling the user to remediate the issue if the cluster already exists. The script that creates the `kind` cluster is from sigstore/scaffolding and I don't see a way to configure it to use a custom name.

File: acceptance/image/image.go [hunk 3]
```diff
@@ -314,6 +316,52 @@ func createAndPushImage(ctx context.Context, imageName string) (context.Context,
...
+	sii, err := layout.SignedImageIndex(imageDir)
+	if err != nil {
+		return ctx, err
+	}
+
+	if err := cosignRemote.WriteSignedImageIndexImages(ref, sii); err != nil {
+		return ctx, err
+	}
+
+	// if state.Images == nil {
```
[the remark this answers is not in the document]
Comment (lcarva) [reply acceptance/image/image.go:356]: Dead code. I don't think we need to worry about the state too much here.

File: acceptance/image/image.go [hunk 4]
```diff
@@ -314,6 +316,52 @@ func createAndPushImage(ctx context.Context, imageName string) (context.Context,
 	return ctx, nil
 }
 
+// TODO: Update description
```
[the remark this answers is not in the document]
Comment (lcarva) [reply acceptance/image/image.go]: Updated the description.

**Pull request merged**